Mathlib/MeasureTheory/Measure/MeasureSpaceDef.lean

```diff
@@ -74,6 +74,54 @@ structure Measure (α : Type*) [MeasurableSpace α] extends OuterMeasure α wher
 /-- Notation for `Measure` with respect to a non-standard σ-algebra in the domain. -/
 scoped notation "Measure[" mα "]" α:arg => @Measure α mα
 
+end MeasureTheory
+
+namespace Mathlib.Meta
+open Lean Elab Term Meta Qq MeasureTheory
+
+/-- Try to elaborate `μ` as a term of type `T` where `OuterMeasureClass T ?Ω`. If that fails, try to
+elaborate `μ` as `Measure ?Ω`. -/
+def elabMeasure (μ : TSyntax `term) (expectedType? : Option Expr) : TermElabM Expr := do
+  let ⟨u, T, _⟩ ← inferTypeQ (← elabTerm μ expectedType?)
+  match u, T with
+  | .succ v, ~q(OuterMeasure $α) => elabTerm μ <| .some q(OuterMeasure $α)
+  | .succ v, _ =>
```

Comment:
You're missing the match for `.zero` here

Reply:
`zero` I just discard, right? How should I handle metavariables? Should I assign them?